spec/basicAuth.spec.ts

```diff
@@ -0,0 +1,97 @@
+import { shallow, ShallowWrapper } from 'enzyme';
+import * as React from 'react';
+import BasicAuth from '../src/renderer/components/basic-auth';
+
+fdescribe('basic auth', () => {
+
+    it('should render correctly', () => {
+        const wrapper: ShallowWrapper = shallow(React.createElement(BasicAuth));
+
+        expect(wrapper).toMatchSnapshot();
+    });
+
+    it('should call `setState` when `change` is call', () => {
+        const spy: jest.SpyInstance = jest.spyOn(BasicAuth.prototype, 'setState');
+        const wrapper: ShallowWrapper = shallow(React.createElement(BasicAuth));
+        const usernameTargetMock: object = { target: { id: 'username', value: 'foo' }};
+        const passwordTargetMock: object = { target: { id: 'password', value: '123456' }};
+
+        wrapper.find('#username').simulate('change', usernameTargetMock);
+
+        expect(spy).lastCalledWith({ username: 'foo' });
+
+        wrapper.find('#password').simulate('change', passwordTargetMock);
+
+        expect(spy).lastCalledWith({ password: '123456' });
+    });
+
+    it('should call submit login', () => {
+        const { ipcRenderer } = require('./__mocks__/electron');
+        const spy: jest.SpyInstance = jest.spyOn(ipcRenderer, 'send');
+        const wrapper: ShallowWrapper = shallow(React.createElement(BasicAuth));
+        const usernameTargetMock: object = { target: { id: 'username', value: 'foo' }};
+        const passwordTargetMock: object = { target: { id: 'password', value: '123456' }};
+
+        wrapper.find('#username').simulate('change', usernameTargetMock);
+        wrapper.find('#password').simulate('change', passwordTargetMock);
+
+        wrapper.find('#basicAuth').simulate('submit');
+
+        expect(spy).lastCalledWith('basic-auth-login', { username: 'foo', password: '123456' });
+    });
+
+    it('should call `basic-auth-closed` event when cancel button is clicked', () => {
+        const { ipcRenderer } = require('./__mocks__/electron');
+        const spy: jest.SpyInstance = jest.spyOn(ipcRenderer, 'send');
+        const wrapper: ShallowWrapper = shallow(React.createElement(BasicAuth));
+
+        wrapper.find('#cancel').simulate('click');
+
+        expect(spy).lastCalledWith('basic-auth-closed', false);
+    });
+
+    describe('basic auth mount and unmount event', () => {
+        const { ipcRenderer } = require('./__mocks__/electron');
+        const basicAuthDataLabel: string = 'basic-auth-data';
+        const onLabelEvent: string = 'on';
+        const removeListenerLabelEvent: string = 'removeListener';
+
+        it('should call `basic-auth-data` event when component is mount', () => {
+            const spy: jest.SpyInstance = jest.spyOn(ipcRenderer, onLabelEvent);
+
+            shallow(React.createElement(BasicAuth));
+
+            expect(spy).toBeCalledWith(basicAuthDataLabel, expect.any(Function));
+        });
+
+        it('should remove listen `basic-auth-data` when component is unmount', () => {
```

Comment:
typo. sentence should be `should remove listen basic-auth-data when component is unmounted`